src/Middleware/RequestThrottling/src/RequestThrottlingMiddleware.cs

```diff
@@ -33,7 +36,14 @@ public RequestThrottlingMiddleware(RequestDelegate next, ILoggerFactory loggerFa
             _next = next;
             _logger = loggerFactory.CreateLogger<RequestThrottlingMiddleware>();
             _options = options.Value;
-            _semaphore = new SemaphoreWrapper(_options.MaxConcurrentRequests); 
+
+            if (_options.MaxConcurrentRequests == null)
+            {
+                // TODO :: Localization
+                throw new ArgumentException(string.Format("Option must be provided {0}", nameof(_options.MaxConcurrentRequests)));
```

Comment:
A few notes:
1. Loc is not really a concern at this point, but it's fine if you want to use a RESX.
2. The exception message itself must be a complete sentence (proper grammar, end with a period, etc.)
3. One of the parameters to `ArgumentException` is the parameter name. In this case the parameter name is `options` (even though the problem is a sub-property of it), so make sure to pass that in.

Reply:
Would this be okay?
```
throw new ArgumentNullException(nameof(_options.MaxConcurrentRequests), "The parameter 'options.MaxConcurrentRequets' must be specified.");
```